src/leap/mail/imap/mailbox.py

```diff
@@ -345,9 +244,10 @@ def getUID(self, message):
 
         :rtype: int
         """
-        msg = self.messages.get_msg_by_uid(message)
-        if msg is not None:
-            return msg.getUID()
+        # TODO --- return the uid if it has it!!!
+        d = self.collection.get_msg_by_uid(message)
+        d.addCallback(lambda m: m.getUID())
+        return d
```

Comment:
I've updated this today. The function now is meaningless (but expected).
For now i'm just returning the UID (there is only one such thing), I'll deal with relative sequence numbers in other moment (TB doesn't seem to use them).